src/metpy/plots/declarative.py

```diff
@@ -494,6 +494,16 @@ def lookup_map_feature(feature_name):
     return feat.with_scale(scaler)
 
 
+def drop_traitlets_dir(self):
+    """Exclude members inherited from HasTraits and '_'/'__' members for dir()."""
+    traits_dir = dir(HasTraits)
+    class_dir = dir(type(self))
+    return [name for name in class_dir if name not in traits_dir if not name.startswith('_')]
+
+
+HasTraits.__dir__ = drop_traitlets_dir
```

Comment:
😬 I really don't want to monkeypatch if we don't have to. We *should* be able to accomplish this by adding:
```python
__dir__ = drop_traitlets_dir
```
inside each of our own base classes?

Reply:
Alternatively, I guess we could do:
```python
class MetpyHasTraits(HasTraits):
    @classmethod
    def __dir__(self):
        ...
```
and have all of ours inherit from that instead of `HasTraits`. That might be cleanest.

Reply:
Yep, I had played around with this. I wasn't sure what preference there'd be for any of these options over others, other than I was avoiding typing `__dir__` a bunch. I like option (b) here and it might make a few other things easier too